### PR TITLE
Delete `GT_STORE_OBJ`

### PR DESCRIPTION
**Pull request by SingleAccretion**
The `STORE` counterpart to #84221.

[No diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=242269&view=ms.vss-build-web.run-extensions-tab), a small TP improvement on 32 bit.

Note: diff best viewed with the "hide whitespaces" option.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/jit-contrib

**Review by jakobbotsch — APPROVED** [review]
Awesome!

File: src/coreclr/jit/gentree.h [hunk 1]
```diff
@@ -7303,6 +7302,10 @@ struct GenTreeBlk : public GenTreeIndir
     enum
     {
         BlkOpKindInvalid,
+        BlkOpKindCpObjUnroll,
+#ifdef TARGET_XARCH
+        BlkOpKindCpObjRepInstr,
+#endif
```
Comment (jakobbotsch) [inline src/coreclr/jit/gentree.h:7308]: These are missing handling in `gtDispTree`, can you submit a fix? (They currently hit an `unreached()`)

Comment (SingleAccretion) [reply src/coreclr/jit/gentree.h:7308]: Working on it.

**Pull request merged**